log/error.go

```diff
@@ -31,94 +34,83 @@ type EventStackTrace struct {
+		evtErrs = append(evtErrs, EventError{
+			Message:    wrappedErr.Error(),
+			StackTrace: extractStacktrace(wrappedErr),
+		})
+	}
+
+	return evtErrs
 }
 
-// FormatErrors returns an option you can pass to Event to attach
-// error information to a log event
-//
-// # It uses error.Error() to stringify the error value
-//
-// It also includes the error type itself as unstructured log
-// data. For a struct{} type, it is included directly. For all
-// other types, it is wrapped in a Data{} struct
-//
-// It also includes a full stack trace to where FormatErrors() is called,
-// so you shouldn't normally store a log.Error for reuse (e.g. as a
-// package level variable)
-func FormatErrors(errs []error) option {
-
-	var e []EventError
-
-	for i := range errs {
-		if errs[i] == nil {
-			continue
-		}
-
-		err := EventError{
-			Message:    errs[i].Error(),
-			StackTrace: make([]EventStackTrace, 0),
-		}
-
-		k := reflect.Indirect(reflect.ValueOf(errs[i])).Type().Kind()
-		switch k {
-		case reflect.Struct:
-
-			// check error types
-			switch newErr := errs[i].(type) {
-			case *CustomError: // matched CustomError type
-				err.Data = newErr.ErrorData()
-			case error:
-				// catch everything else
-			}
-
-		default:
-			// we have something else, so nest it inside a Data value
-			err.Data = Data{"value": errs[i]}
-		}
-
-		pc := make([]uintptr, 10)
-		n := runtime.Callers(2, pc)
-		if n > 0 {
-			frames := runtime.CallersFrames(pc[:n])
-
-			for {
-				frame, more := frames.Next()
-
-				err.StackTrace = append(err.StackTrace, EventStackTrace{
-					File:     frame.File,
-					Line:     frame.Line,
-					Function: frame.Function,
-				})
-
-				if !more {
-					break
-				}
-			}
-		}
-
-		e = append(e, err)
-	}
+func extractStacktrace(err error) []EventStackTrace {
+	st := make([]EventStackTrace, 0)
 
-	a := EventErrors(e)
+	pkg := reflect.Indirect(reflect.ValueOf(err)).Type().PkgPath()
+	switch pkg {
+	case "golang.org/x/xerrors":
+		st = extractXErrStacktrace(err)
+	case "github.com/pkg/errors":
+		st = extractPkgErrStacktrace(err)
+	}
 
-	return &a
+	return st
 }
 
-// CustomError defines an error object that abides to the error type
-// with the extension of including data field
-type CustomError struct {
-	Message string                 `json:"message"`
-	Data    map[string]interface{} `json:"data"`
+func extractXErrStacktrace(xerr error) []EventStackTrace {
+	errstr := fmt.Sprintf("%+v", xerr)
+	lines := strings.Split(errstr, "\n")
+	function := strings.Trim(lines[1], " ")
+	caller := strings.Split(strings.Trim(lines[2], " "), ":")
+	file := caller[0]
+	lineNum, _ := strconv.Atoi(caller[1])
+
+	return []EventStackTrace{{
+		File:     file,
+		Line:     lineNum,
+		Function: function,
+	}}
 }
 
-// Error returns the custom error message embedded in CustomError
-func (c *CustomError) Error() string {
-	return c.Message
+func extractPkgErrStacktrace(pkgerr error) []EventStackTrace {
+	st := []EventStackTrace{}
```

Comment:
I get the error "Empty slice declaration using a literal" here. This solves it:
```suggestion
	var st []EventStackTrace
```

Reply:
Thank you, addressed in https://github.com/ONSdigital/log.go/pull/44/commits/fc70a349237c838e973a6e61addef0ee83910665